adding local Activity in php (#1148)
temporalio/documentation: docs/php/how-to-create-local-activities-in-php.md

---

> id: how-to-create-local-activities-in-php

Reviewer: ```suggestion
id: how-to-execute-a-local-activity-in-php
```

> title: Create local Activities in PHP

Reviewer: ```suggestion
title: How to execute a Local Activity in PHP
```

sidebar_label: Local Activity
---

> To create a local Activity, use `#[LocalActivityInterface]`.

Reviewer: ```suggestion
To execute a Local Activity, use `#[LocalActivityInterface]`.
```

```php
use Temporal\Activity\LocalActivityInterface;

#[ActivityInterface]
interface YourActivityInterface
{
    #[ActivityMethod("youractivity")]
    public function youractivity(): string;
}
```

Local activity requires `LocalActivityOptions`:

```php
$greetingActivity = Workflow::newActivityStub(
    YourActivityInterface::class,
    LocalActivityOptions::new()->withStartToCloseTimeout(\DateInterval::createFromDateString('30 seconds'))
);
```

> Local Activities have a limited set of options, including: `ScheduleToCloseTimeout`, `StartToCloseTimeout`, `RetryPolicy`.

Reviewer 2: Up to you. You're probably sick of my opinions on lists of items.

> Local Activities have a limited set of options, including: `ScheduleToCloseTimeout`, `StartToCloseTimeout`, and `RetryPolicy`.

Reviewer: ```suggestion
Local Activity Execution options include the following: 

- `ScheduleToCloseTimeout`
- `StartToCloseTimeout`
- `RetryPolicy`.
```
